### PR TITLE
Reset password & code restructure

### PR DESCRIPTION
**Pull request by suraj-tekdi**

### CODE REVIEW

File: src/src/services/keycloak/keycloak.service.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public async resetPassword(keycloak_id, token, password) {
+        console.log("resetPassword")
+        const data = {
+            "temporary": false,
+            "type": "password",
+            "value": password
+        }
+
+
+        const url = this.configService.get<string>('KEYCLOAK_URL')+`/admin/realms/eg-sso/users/${keycloak_id}/reset-password`;
```
Comment (manojLondhe) [inline src/src/services/keycloak/keycloak.service.ts:52]: lets add and use config for realm name i.e. `eg-sso`

File: src/src/services/keycloak/keycloak.service.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    constructor(private configService: ConfigService, private readonly httpService: HttpService) { }
+
+    public async getAdminKeycloakToken() {
+        console.log("inside getAdminKeycloakToken")
+        const data = {
+            username: 'admin',
+            client_id: 'admin-cli',
+            grant_type: 'client_credentials',
+            password: this.configService.get<string>('KEYCLOAK_ADMIN_PASSWORD'),
```
Comment (manojLondhe) [inline src/src/services/keycloak/keycloak.service.ts:17]: for `grant_type: 'client_credentials',` you dont need to send password

File: src/src/auth/auth.controller.ts [hunk 3]
```diff
@@ -23,4 +23,10 @@ export class AuthController {
     public verifyOtp(@Body() req: OtpVerifyDTO, @Res() response: Response) {
         return this.authService.verifyOtp(req, response);
     }
+
+    @Post('/reset-password')
+    @UsePipes(ValidationPipe)
+    public resetPassword(@Body() req: any, @Res() response: Response) {
```
Comment (manojLondhe) [inline src/src/auth/auth.controller.ts:29]: what DTO class is applied here?

File: src/src/auth/auth.service.ts [hunk 4]
```diff
@@ -128,4 +132,55 @@ export class AuthService {
 
 
     }
+
+    public async resetPassword(req, response) {
```
Comment (manojLondhe) [inline src/src/auth/auth.service.ts:136]: - we should also get from request mobile, OTP, reason, hash, pwd, confirm pwd all
- validate OTP again
- then change pwd

File: src/src/auth/auth.controller.ts [hunk 5]
```diff
@@ -23,4 +23,10 @@ export class AuthController {
     public verifyOtp(@Body() req: OtpVerifyDTO, @Res() response: Response) {
         return this.authService.verifyOtp(req, response);
     }
+
+    @Post('/reset-password')
```
Comment (manojLondhe) [inline src/src/auth/auth.controller.ts:27]: lets rename route as reset-password-otp
and 
methods as resetPasswordUsingOtp

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/services/keycloak/keycloak.service.ts (hunk 1 above)
Comment (suraj-tekdi) [reply src/src/services/keycloak/keycloak.service.ts]: done

File: src/src/services/keycloak/keycloak.service.ts (hunk 2 above)
Comment (suraj-tekdi) [reply src/src/services/keycloak/keycloak.service.ts]: done

File: src/src/auth/auth.service.ts (hunk 4 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.service.ts]: done

File: src/src/auth/auth.controller.ts (hunk 5 above)
Comment (suraj-tekdi) [reply src/src/auth/auth.controller.ts]: done

**Pull request merged**